client/handle_http.go

```diff
@@ -972,10 +997,9 @@ Loop:
 
 }
 
-var UploadClient = &http.Client{Transport: config.GetTransport()}
-
 // Actually perform the Put request to the server
 func doPut(request *http.Request, responseChan chan<- *http.Response, errorChan chan<- error) {
+	var UploadClient = &http.Client{Transport: config.GetTransport()}
```

Comment:
I moved this within the function because when it was a global variable, the transport was not being set correctly. For example, it did not recognize `TLSSkipVerify` being set to true and it was always false. Somehow it was getting the transport before any `init` functions in config were called.